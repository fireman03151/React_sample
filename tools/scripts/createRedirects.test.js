/* global describe expect */

const { createRedirects } = require('./createRedirects');

const testLocations = {
  api: 'https://api.example.com',
  home: 'https://home.example.com',
  forum: 'https://forum.example.com',
  forumProxy: 'https://proxy.example.com'
};

describe('createRedirects', () => {
  it('is a function', () => {
    expect(typeof createRedirects).toEqual('function');
  });

  it('returns a string', () => {
    expect(typeof createRedirects(testLocations)).toEqual('string');
  });

  it('replaces instances of `#{{...}}` with the locations provided', () => {
    expect.assertions(8);

    const apiPlaceholderRE = /#\{\{API\}\}/;
    const homePlaceholderRE = /#\{\{HOME\}\}/;
    const forumPlacehilderRE = /#\{\{FORUM\}\}/;
    const forumProxyPlaceholderRE = /#\{\{FORUM_PROXY\}\}/;
    const redirects = createRedirects(testLocations);

    const hasApiPlaceholder = apiPlaceholderRE.test(redirects);
    const hasHomePlaceholder = homePlaceholderRE.test(redirects);
    const hasForumPlaceholder = forumPlacehilderRE.test(redirects);
    const hasForumProxyPlaceholder = forumProxyPlaceholderRE.test(redirects);

    expect(hasApiPlaceholder).toBe(false);
    expect(hasHomePlaceholder).toBe(false);
    expect(hasForumPlaceholder).toBe(false);
    expect(hasForumProxyPlaceholder).toBe(false);

    const { api, home, forum, forumProxy } = testLocations;
    expect(redirects.includes(`${api}/internal/:splat`)).toBe(true);
    expect(
      redirects.includes(
        `${home}/forum/t/free-code-camp-privacy-policy/19545 301`
      )
    ).toBe(true);
    expect(redirects.includes(`${forum} 301`)).toBe(true);
    expect(redirects.includes(`${forumProxy} 200`)).toBe(true);
  });

  it('throws when any location is missing', () => {
    expect.assertions(4);

    const api = 'api';
    const home = 'home';
    const forum = 'forum';
    const forumProxy = 'forumProxy';

    const noApi = { forum, home, forumProxy };
    const noHome = { api, forum, forumProxy };
    const noForum = { api, home, forumProxy };
    const noProxy = { api, home, forum };

    expect(() => createRedirects(noApi)).toThrow();
    expect(() => createRedirects(noHome)).toThrow();
    expect(() => createRedirects(noForum)).toThrow();
    expect(() => createRedirects(noProxy)).toThrow();
  });

  it('matches the snapshot', () =>
    expect(createRedirects(testLocations)).toMatchSnapshot());
});
