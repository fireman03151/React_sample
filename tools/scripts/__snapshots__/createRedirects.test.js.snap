// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRedirects matches the snapshot 1`] = `
"#api redirect

/internal/*                                   https://api.example.com/internal/:splat 200

# auth redirects
/signin                                       https://api.example.com/signin 200
/signup                                       https://api.example.com/signin 200
/email-signin                                 https://api.example.com/signin 200
/login                                        https://api.example.com/signin 200
/deprecated-signin                            https://api.example.com/signin 200
/logout                                       https://api.example.com/signout 200
/passwordless-change                          https://api.example.com/confirm-email 200

# certification redirects
/:username/front-end-certification            /certification/:username/legacy-front-end 301
/:username/data-visualization-certification   /certification/:username/legacy-data-visualization 301
/:username/back-end-certification             /certification/:username/legacy-back-end 301
/:username/full-stack-certification           /certification/:username/full-stack 301

# unsunscribe redirects
/u/*                                          https://api.example.com/u/:splat 200
/unsunscribe/*                                https://api.example.com/unsunscribe/:splat 200
/ue/*                                         https://api.example.com/ue/:splat 200

# misc redirects
/agile                                        / 301
/chat                                         https://gitter.im/FreeCodeCamp/FreeCodeCamp 301
/twitch                                       https://twitch.tv/freecodecamp 301
/nonprofits-form                              / 301
/pmi-acp-agile-project-managers               / 301
/pmi-acp-agile-project-managers-form          / 301
/stories                                      / 301
/all-stories                                  / 301
/field-guide/*                                /forum 301
/learn-to-code                                /learn 200
/map                                          /learn 200
/news                                         https://news.example.com 200
/news/*                                       https://news.example.com/:splat 200
/forum/*                                      https://forum.example.com/:splat 200
/privacy                                      https://forum.example.com/t/free-code-camp-privacy-policy/19545 301
/nonprofit-project-instructions               https://forum.example.com/t/how-free-code-camps-nonprofits-projects-work/19547 301
/how-nonprofit-projects-work                  https://www.freecodecamp.org/news/open-source-for-good-1a0ea9f32d5a 301

"
`;
